packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -242,21 +263,39 @@ class RequestWrapper {
    * @param {Request} [input.cacheKey] Supply a cacheKey if you wish to cache
    *        the response against an alternative request to the `request`
    *        argument.
+   * @param {function} [input.handlerDefaultCacheableResponsePlugin] Allows the
+   *        caller to override the default check for cacheability, for
+   *        situations in which the cacheability check wasn't explicitly
+   *        configured when constructing the `RequestWrapper`.
    * @return {Promise.<Response>} The network response.
    */
-  async fetchAndCache({request, waitOnCache, cacheKey}) {
+  async fetchAndCache({request, waitOnCache, cacheKey,
+                        handlerDefaultCacheableResponsePlugin}) {
     assert.atLeastOne({request});
 
     let cachingComplete;
     const response = await this.fetch({request});
 
-    // response.ok is true if the response status is 2xx.
-    // That's the default condition.
-    let cacheable = response.ok;
-    if (this.plugins.has('cacheWillUpdate')) {
-      const plugin = this.plugins.get('cacheWillUpdate')[0];
-      cacheable = plugin.cacheWillUpdate({request, response});
-    }
+    // We need flexibility in determining whether a given response should
+    // be added to the cache. There are several possible ways that this logic
+    // might be specified, and they're given the following precedence:
+    // 1. Passing in a `CacheableResponsePlugin` to the `RequestWrapper`
+    //    constructor, which sets this._userSpecifiedCachableResponsePlugin.
+    // 2. Passing in a parameter to the fetchAndCache() method (done by certain
+    //    runtime handlers, like `StaleWhileRevalidate`), which sets
+    //    handlerDefaultCacheableResponsePlugin.
+    // 3. The default that applies to anything using the `RequestWrapper` class
+    //    that doesn't specify the custom behavior, which is accessed via
+    //    the this.getDefaultCacheableResponsePlugin().
+    const effectiveCacheableResponsePlugin =
+      this._userSpecifiedCachableResponsePlugin ||
+      handlerDefaultCacheableResponsePlugin ||
```

Comment:
Could this just be named cacheResponsePlugin. I can't think of a better way to structure this code, but multiple "defaults" is bugging me.